src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const onLikeBtnClick = () => {
+    const updatedEntry = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked
+    };
+
+    props.onUpdate(updatedEntry);
+  };
```

Comment:
Just to reiterate, we don't need to pass the entire entry to `onUpdate` since we can use the state set in `App.js`. 
Here's how that function would look in its utilization:
```JSX
 <button className="like" onClick={() => props.onUpdate(props.id)}>{props.liked ? '❤️' : '🤍'}</button>
```